core/services/ocr2/plugins/ocr2keeper/evmregistry/v20/registry.go

```diff
@@ -716,7 +716,7 @@ func (r *EvmRegistry) getUpkeepConfigs(ctx context.Context, ids []*big.Int) ([]a
 	)
 
 	for i, id := range ids {
-		opts, err := r.buildCallOpts(ctx, nil)
+		opts, err := r.buildCallOpts(ctx, big.NewInt(0))
```

Comment:
should this change be done within `buildCallOpts` so that if other services are using the function they don't get a nil pointer?

maybe just the if condition in line 469 can be removed

Reply:
Maybe! I wanted to see what happens with just this change so far; buildCallOpts has been updated to initialise with whatever block we pass rather than directly with nil, and I wanted to limit the change for now so that we pass a non-nil value here.

Reply:
I'm working off the assumption that specifying block 0 actually means the latest block is used - is that correct?

Reply:
2.0 upgrade passes: https://github.com/smartcontractkit/chainlink/actions/runs/9744742196/job/26891191178

Reply:
Updated PR still passing with 2.0 upgrade: https://github.com/smartcontractkit/chainlink/actions/runs/9746647010